src/FsUnit.NUnit/FsUnitTyped.fs

```diff
@@ -5,15 +5,17 @@ open NUnit.Framework
 open System.Collections.Generic
 
 [<AutoOpen>]
-
 module TopLevelOperators =
+
     [<DebuggerStepThrough>]
     let shouldEqual (expected: 'a) (actual: 'a) =
-        Assert.That(actual, FsUnit.Equality.IsEqualTo(expected), sprintf "Expected: %A\nActual: %A" expected actual)
+        FsUnit.TopLevelOperators.FSharpCustomMessageFormatter() |> ignore
```

Comment:
I believe that we should do it only once, but not on every call...
Right now it looks like we call `TestContext.AddFormatter(ValueFormatterFactory(fun _ -> ValueFormatter(sprintf "%A")))` on every assertion.

Reply:
Do you have an idea how to implicitly call it?

Reply:
No, we have to do it explicitly, bug we can do it using `lazy` and ensure that only one `ValueFormatter` is registered in `TestContext`

Reply:
If we want to support `FSharpCustomMessageFormatter` in existing solution, then we should add static lazy coda that register formatter and force it from FSharpCustomMessageFormatter constructor, `shouldEqual` and `shouldNotEqual`

https://github.com/fsprojects/FsUnit/blob/67dcc91f8b5cf60b1b80fb3a63621136f9df9344/src/FsUnit.NUnit/FsUnit.fs#L13

Reply:
Unfortunately I'm not able to do it the `lazy` way or I'm doing it wrong. :(
Do you have a reference how to do it properly?

Reply:
We may have to add this formatter in every context in NUnit.
I tested: `[ Ok "ok" ] |> should be (supersetOf [ Ok "no" ])` 
It is printed ugly aswell. When I add `FSharpCustomMessageFormatter |> ignore` to `FsUnit.should` then it's pretty.

Reply:
I don't really know how we go on now.